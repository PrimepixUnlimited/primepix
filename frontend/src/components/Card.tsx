import React, { FC } from 'react'
import { StyleSheet } from 'react-native'
import { Card, CardProps } from 'react-native-elements'
import styles from '../constants/styles'

interface Props extends CardProps {
  // custom props
}

const CardWrapper: FC<Props> = ({
  children,
  containerStyle = s.container,
  dividerStyle,
  featuredSubtitle,
  featuredSubtitleStyle,
  featuredTitle,
  featuredTitleStyle,
  image,
  imageProps,
  imageStyle,
  imageWrapperStyle,
  title,
  titleStyle = s.title,
  wrapperStyle
}) => (
  <Card
    containerStyle={containerStyle}
    dividerStyle={dividerStyle}
    featuredSubtitle={featuredSubtitle}
    featuredSubtitleStyle={featuredSubtitleStyle}
    featuredTitle={featuredTitle}
    featuredTitleStyle={featuredTitleStyle}
    image={image}
    imageProps={imageProps}
    imageStyle={imageStyle}
    imageWrapperStyle={imageWrapperStyle}
    title={title}
    titleStyle={titleStyle}
    wrapperStyle={wrapperStyle}
  >
    {children}
  </Card>
)

const s = StyleSheet.create({
  container: {
    backgroundColor: styles.greyScale.black1,
    borderWidth: 0,
    borderRadius: 10
  },
  title: {
    color: styles.greyScale.white,
    fontSize: 22,
    fontWeight: 'normal',
    textAlign: 'left'
  }
})

export default CardWrapper
