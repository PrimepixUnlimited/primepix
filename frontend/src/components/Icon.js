import React from 'react';
import {Icon} from 'react-native-elements';

const IconWrapper = ({
  color,
  containerStyle,
  name,
  onPress,
  size,
  type,
  underlayColor,
}) => (
  <Icon
    color={color}
    containerStyle={containerStyle}
    name={name}
    onPress={onPress}
    size={size}
    type={type}
    underlayColor={underlayColor}
  />
);

export default IconWrapper;
