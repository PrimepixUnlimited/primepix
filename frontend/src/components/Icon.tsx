import React, { FC } from 'react'
import { Icon, IconProps } from 'react-native-elements'

interface Props extends IconProps {
  // custom props
}

const IconWrapper: FC<Props> = ({
  color,
  containerStyle,
  name,
  onPress,
  size,
  type,
  underlayColor
}) => (
  <Icon
    color={color}
    containerStyle={containerStyle}
    name={name}
    onPress={onPress}
    size={size}
    type={type}
    underlayColor={underlayColor}
  />
)

export default IconWrapper
