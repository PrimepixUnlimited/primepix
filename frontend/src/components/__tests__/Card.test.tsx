// components/__tests__/Hello.tsx
import 'react-native'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Card from '../Card'
import Icon from '../Icon'

describe('Card snapshot', () => {
  it('renders correctly', () => {
    const card = renderer
      .create(
        <Card
          featuredSubtitle="Featured Subtitle"
          featuredTitle="Featured Title"
          title="Title"
        />
      )
      .toJSON()
    expect(card).toMatchSnapshot()
  })
})
