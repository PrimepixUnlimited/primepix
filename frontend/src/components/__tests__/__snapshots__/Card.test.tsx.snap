// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#18191C",
      "borderColor": "#e1e8ee",
      "borderRadius": 10,
      "borderWidth": 0,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "normal",
            "marginBottom": 15,
            "textAlign": "left",
          }
        }
        testID="cardTitle"
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Title
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#bcbbc1",
            "height": 0.5,
            "marginBottom": 15,
          }
        }
      />
    </View>
  </View>
</View>
`;
